contracts/rootchain/RootChain.sol

```diff
@@ -149,29 +150,32 @@ contract RootChain is Ownable {
     // @param confirmSignatures confirm signatures sent by the owners of the inputs acknowledging the spend.
     // @notice `confirmSignatures` and `ConfirmSig0`/`ConfirmSig1` are unrelated to each other.
     // @notice `confirmSignatures` is either 65 or 130 bytes in length dependent on if input2 is used.
-    function startTransactionExit(uint256[3] txPos, bytes txBytes, bytes proof, bytes sigs, bytes confirmSignatures)
+    function startTransactionExit(uint256[3] txPos, bytes txBytes, bytes proof, bytes confirmSignatures)
         public
         payable
     {
-        RLPReader.RLPItem[] memory txList = txBytes.toRlpItem().toList();
-        require(txList.length == 17, "incorrect tx length");
+        RLPReader.RLPItem[] memory txList;
+        bytes[] memory sigList;
+        bytes memory encodedTxList;
+        (txList, sigList, encodedTxList) = extractTxListAndSigs(txBytes);
+
         require(msg.sender == txList[12 + 2 * txPos[2]].toAddress(), "caller must own this utxo");
         require(msg.value >= minExitBond, "insufficient exit bond");
         if (msg.value > minExitBond) {
-            uint256 excess = msg.value.sub(minExitBond);
-            balances[msg.sender] = balances[msg.sender].add(excess);
-            totalWithdrawBalance = totalWithdrawBalance.add(excess);
+            /* uint256 excess = msg.value.sub(minExitBond); */
+            balances[msg.sender] = balances[msg.sender].add(msg.value.sub(minExitBond));
+            totalWithdrawBalance = totalWithdrawBalance.add(msg.value.sub(minExitBond));
```

Comment:
I think we should store the excess amount in variable rather than redoing the same computation. Are you getting a stack too deep error for having the local variable?

Reply:
Yeah that's the main issue